src/layers/ZoomLayer.ts

```diff
@@ -8,18 +8,19 @@ export class ZoomLayer extends OptionalLayer {
     super();
   };
 
-  private handleZoom = (zoomExtents: ZoomExtents, layerArgs: LayerArgs) => {
+  private handleZoom = async (zoomExtents: ZoomExtents, layerArgs: LayerArgs) => {
     const { x: scaleX, y: scaleY } = layerArgs.scaleConfig.linearScales;
+
+    layerArgs.optionalLayers.forEach(layer => layer.beforeZoom(zoomExtents));
```

Comment:
it may eventually be useful to make this async but I guess that's not needed yet.